modules/standard/Reflection.chpl

```diff
@@ -75,7 +73,32 @@ proc getField(x:?t, param s:string) {
   return __primitive("field value by num", x, i);
 }
 
-// TODO ref return version of getField(x, string)
+/* Get a mutable ref to the ith field in a class or record.
+   Causes a compilation error if `i` is not in 1..numFields(t)
+   or if the argument is not mutable.
+
+   :arg x: a class or record
+   :arg i: which field to get
+   :returns: an rvalue referring to that field.
+ */
+inline
+proc getFieldRef(ref x:?t, param i:int) ref
+  return __primitive("field value by num", x, i);
```

Comment:
Really minor, but in retrospect, it's too bad we called this primitive "field *value* by num" given that refs and values are often in opposition to one another (i.e., it seems slightly counterintuitive that to get a reference to the field, we're asking for its value).  You're obviously inheriting this, but if you wanted to houseclean, changing this name to simply "field by num" or "get field by num" seem better for the value and ref contexts.